### PR TITLE
Feature/215 add democracy dashboard teaser

### LINKED ISSUE #215: Feature: Add Democracy section to the dashboard
The following data can be displayed in the dashboard:

Screenshot 2021-07-23 at 2 50 33 PM

- [ ] Remove the Referenda section from the dashboard and make the new Democracy section to navigate to the referenda screen (similar to the Treasury section and the proposals).

### PR DESCRIPTION
**Pull request by ahoseinian**
### Description
- Show teaser for democracy
Screenshot 2021-07-27 at 15 12 34

- Clears the query cache after changing networks to avoid showing invalid data from other networks

fixes #215

### CODE REVIEW

File: src/api/hooks/useDemocracy.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import useApiQuery from 'src/api/hooks/useApiQuery';
+
+export function useDemocracy() {
+  return useApiQuery(['democracy'], async (api) => {
+    const [referndums, activeProposals, publicPropCount, referendumTotal] = await Promise.all([
```
Comment (silva-fj) [inline src/api/hooks/useDemocracy.ts]:
```suggestion
    const [referendums, activeProposals, publicPropCount, referendumTotal] = await Promise.all([
```

File: src/api/hooks/useVotingStatus.ts [hunk 2]
```diff
@@ -79,11 +80,8 @@ export function useVotingStatus(
 }
 
 export function useBestNumber(): BlockNumber | undefined {
```
Comment (silva-fj) [inline src/api/hooks/useVotingStatus.ts]: Can you move this hook to its own file?

File: src/layout/DemocracySummaryTeaser.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import {formatNumber, BN_ONE, BN_ZERO, BN_HUNDRED} from '@polkadot/util';
+import Padder from 'presentational/Padder';
+import {useBestNumber} from 'src/api/hooks/useVotingStatus';
+import ProgressChartWidget from 'presentational/ProgressWidget';
+import {useBlockTime} from 'src/api/hooks/useBlockTime';
+import {standardPadding} from 'src/styles';
+import {notEmpty} from 'src/utils';
+
+type Props = {
+  onMorePress: () => void;
```
Comment (silva-fj) [inline src/layout/DemocracySummaryTeaser.tsx]: NIT: `onPressMore` or `onMorePressed`

File: src/layout/DemocracySummaryTeaser.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  const timeLeft = total?.sub(progress);
+
+  const progressPercent = progress
+    .mul(BN_HUNDRED)
+    .div(total ?? BN_ONE)
+    .toNumber();
+
+  const {timeStringParts} = useBlockTime(timeLeft);
+
+  const firstTwoNoneEmptyTimeParts = timeStringParts.filter(notEmpty).slice(0, 2);
```
Comment (silva-fj) [inline src/layout/DemocracySummaryTeaser.tsx]: You can also do:
```
  const firstTwoNoneEmptyTimeParts = timeStringParts.filter(Boolean).slice(0, 2);
```

File: src/utils.ts [hunk 5]
```diff
@@ -200,3 +200,8 @@ export function validateFormField(
       !notEnding.some((check) => value.endsWith(check)))
   );
 }
+
+// Remove all empty values from array
```
Comment (silva-fj) [inline src/utils.ts]: What array? :D

**Review by silva-fj — DISMISSED** [review]

File: src/api/hooks/useVotingStatus.ts (hunk 2 above)
Comment (ahoseinian) [reply src/api/hooks/useVotingStatus.ts]: done ✅

File: src/layout/DemocracySummaryTeaser.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/layout/DemocracySummaryTeaser.tsx]: done ✅

File: src/layout/DemocracySummaryTeaser.tsx (hunk 4 above)
Comment (ahoseinian) [reply src/layout/DemocracySummaryTeaser.tsx]: yes makes more sense, changed to this.

File: src/utils.ts (hunk 5 above)
Comment (ahoseinian) [reply src/utils.ts]: I kept this for future use, but made the comment more sensible :D

File: src/layout/CouncilSummaryTeaser.tsx [hunk 6]
```diff
@@ -11,15 +11,15 @@ import ProgressChartWidget from 'presentational/ProgressWidget';
...
+  onPressMore: () => void;
 };
 
 function CouncilSummaryTeaser(props: PropTypes & InjectedPropTypes) {
   const {timeStringParts} = useBlockTime(props.electionsInfo.data.termDuration);
   const {timeStringParts: termLeft} = useBlockTime(props.electionsInfo.data.termLeft);
 
   return (
-    <SeactionTeaserContainer onMorePress={props.onMorePress} title="Council">
+    <SeactionTeaserContainer onMorePress={props.onPressMore} title="Council">
```
Comment (silva-fj) [inline src/layout/CouncilSummaryTeaser.tsx]:
```suggestion
    <SeactionTeaserContainer onPressMore={props.onPressMore} title="Council">
```

**Review by silva-fj — DISMISSED** [review]

File: src/layout/CouncilSummaryTeaser.tsx (hunk 6 above)
Comment (ahoseinian) [reply src/layout/CouncilSummaryTeaser.tsx]: done ✅

**Review by silva-fj — APPROVED** [review]

**Pull request merged**